src/components/SignupScreen.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import {addUser} from '../store/UsersActions';
+
+import {
+  Button,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+
+class SignupScreen extends Component {
+  constructor(){
+    super();
+    this.state = {
+      email: "",
+      password: "",
+      confirmPassword: "",
+    }
+    this.onSignUp = this.onSignUp.bind(this);
+  }
+
+  onSignUp(navigation){
+    const expression = /^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$/;
+    if(this.state.password.length > 8 && expression.test(this.state.email.toLowerCase())){
+      if (this.state.password == this.state.confirmPassword){
+        this.props.onSignUp(this.state.email);
+        navigation.navigate('Details');
+      }
+      else{
+        alert("Different passwords");
+      }
+    }
+    else{
+      alert("Invalid email or password");
+    }
+  }
+
+  render(){
+    const { navigation } = this.props;
+    return (
+      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
```

Comment:
Já que temos um objeto do style podemos passar para la estes estilos e evitamos ter estilos inline.
```suggestion
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
```